datalad/config.py

```diff
@@ -21,13 +22,7 @@
 
 import re
 import os
-from os.path import (
-    join as opj,
-    exists,
-    getmtime,
-    abspath,
-    isabs,
-)
+from pathlib import Path
```

Comment:
General remark: It seems quite inconsistent to me, where we import `Path`from. Half the code base gets it from `pathlib` and the other half from `datalad.utils`.

Reply:
It is of value to keep the dependencies of this code to the rest of datalad at the minimum level. Unless there is a reason to enforce consistency, I'd rather not be consistent here.

Reply:
The only reason is, that there's hardly a point in centralizing the import from pathlib in the first place, if we don't stick to it. Not a merge blocker though.

Reply:
I believe it is OK to not be consistent for a reason. And the reason here is that ConfigManager is needed very early in DataLad and the more of DataLad it needs to import, the more difficult it becomes to make changes in the rest of DataLad.